QuizApp/QuizApp/Coordinator.swift

```diff
@@ -2,19 +2,27 @@ import UIKit
 
 class Coordinator: CoordinatorProtocol {
 
-    private let appDependencies = AppDependencies()
+    private let appDependencies: AppDependencies
     private let navigationController: UINavigationController
 
-    init(navigationController: UINavigationController) {
+    init(navigationController: UINavigationController, appDependencies: AppDependencies) {
         self.navigationController = navigationController
+        self.appDependencies = appDependencies
     }
 
     func showLogIn() {
         let logInViewController = LoginViewController(
             viewModel: LoginViewModel(
-                loginUseCase: appDependencies.loginUseCase))
+                loginUseCase: appDependencies.loginUseCase,
+                coordinator: self))
 
         navigationController.pushViewController(logInViewController, animated: true)
     }
 
+    func showUserViewController() {
+        let userViewController = UserViewController()
+
+        navigationController.pushViewController(userViewController, animated: true)
```

Comment:
You get to UserVC from LoginVC, which means that this VC will be pushed to the navigation stack. Is that correct or should this VC be the only one in the stack?